### PR TITLE
core: Add PDF content validator [SFEQS-1110]

### PR DESCRIPTION
**Pull request by grausof**
This PR adds the validation of the pdf files uploaded through the [pdf-lib](https://pdf-lib.js.org/) library.

#### Motivation and Context
Currently the file upload flow does not provide any control on the type and / or format therefore it is possible to upload any type of file without any control. File format control is added with this PR.

#### How Has This Been Tested?
Sested on simulator

#### Types of changes
- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: src/infra/azure/storage/document.ts [hunk 1]
```diff
@@ -25,3 +27,14 @@ export const makeDeleteDocumentUploaded: R.Reader<
...
   pipe(client, selectBlob(documentId), blobDelete);
+
+export const makeDownloadDocumentUploaded: R.Reader<
+  ContainerClient,
+  DownloadUploadDocumentFromBlob
+> = (client) => (documentId) =>
+  pipe(client, selectBlob(documentId), (blob) =>
+    TE.tryCatch(
+      () => blob.downloadToBuffer(),
+      () => new Error("The blob cannot be downloaded.")
```
Comment (gunzip) [inline src/infra/azure/storage/document.ts]: remember to log the original error at least

File: src/app/use-cases/validate-pdf.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { pipe } from "fp-ts/function";
+
+export const getPdfMetadata = (buffer: Buffer) =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        PDFDocument.load(buffer, {
+          updateMetadata: false,
+        }),
+      () => new Error("This is not a valid PDF")
```
Comment (gunzip) [inline src/app/use-cases/validate-pdf.ts]: can we keep some explicit error here? ie.

Comment (gunzip) [inline src/app/use-cases/validate-pdf.ts]:
```suggestion
      toError
```

File: src/app/use-cases/status-document.ts [hunk 3]
```diff
@@ -53,6 +53,8 @@ const whenValidationInProgress =
       case "MARK_AS_READY":
         return E.right({ ...document, status: "READY" });
       case "MARK_AS_INVALID":
+        // eslint-disable-next-line functional/immutable-data
```
Comment (gunzip) [inline src/app/use-cases/status-document.ts:56]:
```suggestion
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const { url, ...doc } = document;
        return E.right({ ...doc, status: "VALIDATION_ERROR" });
```

Comment (grausof) [reply src/app/use-cases/status-document.ts:56]: I get the following error: `'url' is assigned a value but never used.`

File: src/app/use-cases/validate-pdf.ts (hunk 2 above)
Comment (grausof) [reply src/app/use-cases/validate-pdf.ts]: 👍

File: src/app/use-cases/status-document.ts (hunk 3 above)
Comment (gunzip) [reply src/app/use-cases/status-document.ts:56]: I've updated the suggestion

**Pull request closed without merging**